deepforest/cascade.py

```diff
@@ -762,7 +763,8 @@ def n_aug_features_(self):
 
     # flake8: noqa: E501
     def fit(self, X, y, sample_weight=None):
-
+        X = check_array(X)
```

Comment:
So `check_array` from scikit-learn will automatically cast the dataframe into the numpy array?

Reply:
Yes. A similar method is used in https://github.com/stanfordmlgroup/ngboost/blob/master/ngboost/api.py#300, and I believe it should be right.

Reply:
I will try out the model built on this PR when I get a moment. Thanks!